ecommerce_frontend/src/Pages/Home.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect,useState } from 'react';
+import Navbar from '../Components/Navbar';
+import axios from 'axios';
+import Cookies from 'js-cookie';
+
+export default function Home() {
+  const [userName ,setUserName]=useState(null);
+  const [userEmail ,setUserEmail]=useState(null);
+
+  const authenticate_token = async () => {
+    alert("token verification function called ");
+    try {
+      const response = await axios.get("http://localhost:3001/validate_token", {
```

Comment:
Do not hard code URLs or the base domain; use ENV for the base domain and separate files for APIs, where you can define APIs using functions that take args and return an API route for you to use.

Reply:
like this :

REACT_APP_API_BASE_URL=http://localhost:3000/api
,

const API_BASE_URL = process.env.REACT_APP_API_BASE_URL;
export const getApiUrl = (endpoint) => `${API_BASE_URL}${endpoint}`;
// Example API routes
export const USERS_API = getApiUrl('/users');
export const PRODUCTS_API = getApiUrl('/products');
export const ORDERS_API = getApiUrl('/orders');

// Function for dynamic endpoints
export const userDetailApi = (userId) => getApiUrl(`/users/${userId}`);
export const productDetailApi = (productId) => getApiUrl(`/products/${productId}`);